src/interface_py/h2o4gpu/solvers/truncated_svd.py

```diff
@@ -82,10 +96,15 @@ def fit_transform(self, X, y=None):
         param.k = self.n_components
         param.algorithm = self.algorithm.encode('utf-8')
         param.tol = self.tol
+        param.n_iter = self.n_iter
+        param.random_state = self.random_state
+        param.verbose = 1 if self.verbose else 0
         param.gpu_id = self.gpu_id
 
-        if param.tol <= 0.0:
-            raise ValueError("The `tol` parameter must be > 0.0")
+        if param.tol < 0.0:
+            raise ValueError("The `tol` parameter must be >= 0.0")
```

Comment:
Tol should be > 0.0

Reply:
It's on purpose.  They can choose 0.0 if they want it to run effectively forever, like explained in params.